### PR TITLE
test(action-popover): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor for `action-popover` component to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA
- [ ] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Run npx cypress open-ct to check if there is newly added test.file
 - [x] Check if the `action-popover.test.js` file passed
 - [x] Compare all tests were moved/refactored from .feature file -> .test.js file
 - [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/action-popover/action-popover.test.js [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+
+    it.each([
+      [0, "Business"],
+      [1, "Email Invoice"],
+      [2, "Print Invoice"],
+      [3, "Download PDF"],
+      [4, "Download CSV"],
+      [5, "Delete"],
+    ])(
+      "should render ActionPopover and be able to press Tab %s times and get button %s focused",
```
Comment (sianford) [inline src/components/action-popover/action-popover.test.js:229]: Should read "be able to press down arrow %s times" not Tab as Tab would take the user out of the extended popover.

File: src/components/action-popover/action-popover.test.js [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+      CypressMountWithProviders(
+        <ActionPopoverMenuWithProps focusIndex={null} />
+      );
+
+      actionPopoverButton().eq(0).click();
+      actionPopover().eq(1).children().eq(0).should("not.be.focused");
+    });
+  });
+
+  describe("Check events for ActionPopover component", () => {
```
Comment (sianford) [inline src/components/action-popover/action-popover.test.js:638]: Should we also test the onOpen and onClose (popover opened and popover closed) events?

File: src/components/action-popover/action-popover.test.js (hunk 1 above)
Comment (DlgSHi) [reply src/components/action-popover/action-popover.test.js]: updated

File: src/components/action-popover/action-popover.test.js [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+    });
+
+    it("should open ActionPopover and close it by clicking onto Open icon", () => {
+      CypressMountWithProviders(<ActionPopoverCustom />);
+
+      actionPopoverButton().eq(0).dblclick();
+      actionPopover().should("not.exist");
+    });
+
+    it("should open ActionPopover and close it by clicking onto Open icon", () => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply src/components/action-popover/action-popover.test.js]: deleted

File: src/components/action-popover/action-popover.test.js [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+    it("should render ActionPopoverMenu with menuID", () => {
+      CypressMountWithProviders(
+        <ActionPopoverMenuWithProps menuID="cypress" />
+      );
+
+      actionPopoverButton().eq(0).click();
+      actionPopover().eq(1).should("have.attr", "id", "cypress");
+    });
+
+    it("should render ActionPopoverMenu with focusIndex", () => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply src/components/action-popover/action-popover.test.js]: updated

File: src/components/action-popover/action-popover.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/action-popover/action-popover.test.js:642]: added

**Review by sianford — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Pull request merged**